modules/shortcodes/tweet.php

```diff
@@ -142,4 +143,61 @@ static public function jetpack_tweet_shortcode_script() {
 		}
 	}
 
+	static public function enqueue_block_editor_assets() {
+		wp_enqueue_script( 'wp-blocks' );
+		wp_enqueue_script( 'wp-i18n' );
+		wp_enqueue_script( 'wp-element' );
+		wp_enqueue_script( 'shortcode' );
+		add_action( 'admin_print_footer_scripts', array( __CLASS__, 'admin_footer' ), 999 );
+	}
+
+	static public function admin_footer() {
+?>
+<script>
+( function( wp ) {
+	var blockStyle = {
+			backgroundColor: '#900',
+			color: '#fff',
+			padding: '20px'
+		};
+
+	wp.blocks.registerBlockType( 'jetpack/tweet', {
+		title: wp.i18n.__( 'Tweet', 'jetpack' ),
+		icon: 'twitter',
+		category: 'layout',
+
+		attributes : {
+			tweet : wp.blocks.query.query( 'input[name=tweet]').value
```

Comment:
Currently this is possible by defining `attributes` as a function, which is passed the raw content (string) of the block and expected to return an object of attributes.

https://github.com/WordPress/gutenberg/blob/33542dd/blocks/api/parser.js#L28-L29
https://github.com/WordPress/gutenberg/blob/master/blocks/README.md#wpblocksregisterblocktype-name-string-typedefinition-object-

However, with increasing pressure to introduce server-capable attributes schemas / parsing, I'm not sure whether this can be relied upon.

Some more discussion at: https://github.com/WordPress/gutenberg/pull/1905